src/main.cpp

```diff
@@ -412,138 +534,116 @@ GbdxmArgs* readPackArgs(const po::variables_map& vm)
         errors.push_back("Missing required metadata arguments: --" + join(cliFields, ", --"));
     }
 
-    if(!errors.empty()) {
-        for(const auto& error : errors) {
-            cerr << error << endl;
+    auto categories = args->identifier->detectCategory(*args->package);
+    if(categories.empty()) {
```

Comment:
Consider adding a case when a category is supplied and the model can not be detected.  Saying the "category could not be detected" is confusing (since the user is giving it!).
```
if(categories.empty() && !metadata.category().empty()) {
    errors.push_back("Invalid or unsupported model.");
} else ...
```